File: semana 3/Programación Tradicional.py
#Programación Tradicional (Estructurada)
#Este enfoque se basa en funciones que realizan tareas específicas
#El código se organiza de manera secuencial y lógica, separando la entrada de datos del cálculo

def ingresar_temperaturas_diarias():
    """
    Solicita al usuario las temperaturas de los 7 días de la semana.
    """
    temperaturas = []
    dias = ["lunes", "martes", "miércoles", "jueves", "viernes", "sábado", "domingo"]
    for dia in dias:
        while True:
            try:
                temp = float(input(f"Ingrese la temperatura para el {dia}: "))
                temperaturas.append(temp)
                break
            except ValueError:
                print("Por favor, ingrese un número válido.")
    return temperaturas

def calcular_promedio_semanal(temperaturas):
    """
    Calcula el promedio de una lista de temperaturas.
    """
    if not temperaturas:
        return 0
    return sum(temperaturas) / len(temperaturas)

def main_tradicional():
    """
    Función principal que organiza la ejecución del programa.
    """
    print("--- Calculadora de Promedio del Clima (Tradicional) ---")
    temperaturas_semana = ingresar_temperaturas_diarias()
    promedio = calcular_promedio_semanal(temperaturas_semana)
    print("\n----------------------------------------------------")
    print(f"El promedio semanal de la temperatura es: {promedio:.2f}°C")
    print("----------------------------------------------------")

main_tradicional()

File: semana 3/programación tradicional.java
import java.util.ArrayList;
import java.util.InputMismatchException;
import java.util.List;
import java.util.Scanner;

public class ClimaTradicional {

    /**
     * Solicita al usuario las temperaturas de los 7 días de la semana.
     * @param scanner El objeto Scanner para leer la entrada del usuario.
     * @return Una lista de las temperaturas ingresadas.
     */
    public static List<Double> ingresarTemperaturasDiarias(Scanner scanner) {
        List<Double> temperaturas = new ArrayList<>();
        String[] dias = {"lunes", "martes", "miércoles", "jueves", "viernes", "sábado", "domingo"};

        for (String dia : dias) {
            while (true) {
                try {
                    System.out.print("Ingrese la temperatura para el " + dia + ": ");
                    double temp = scanner.nextDouble();
                    temperaturas.add(temp);
                    break; // Sale del bucle si la entrada es válida
                } catch (InputMismatchException e) {
                    System.out.println("Error: Por favor, ingrese un número válido.");
                    scanner.next(); // Limpia el buffer del scanner
                }
            }
        }
        return temperaturas;
    }

    /**
     * Calcula el promedio de una lista de temperaturas.
     * @param temperaturas La lista de temperaturas.
     * @return El promedio calculado.
     */
    public static double calcularPromedioSemanal(List<Double> temperaturas) {
        if (temperaturas == null || temperaturas.isEmpty()) {
            return 0.0;
        }
        double suma = 0.0;
        for (double temp : temperaturas) {
            suma += temp;
        }
        return suma / temperaturas.size();
    }

    /**
     * Función principal que organiza la ejecución del programa.
     */
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("--- Calculadora de Promedio del Clima (Tradicional - Java) ---");
        List<Double> temperaturasSemana = ingresarTemperaturasDiarias(scanner);
        double promedio = calcularPromedioSemanal(temperaturasSemana);

        System.out.println("\n----------------------------------------------------");
        // Usamos printf para formatear la salida a dos decimales
        System.out.printf("El promedio semanal de la temperatura es: %.2f°C\n", promedio);
        System.out.println("----------------------------------------------------");

        scanner.close(); // Buena práctica cerrar el scanner
    }
}